### PR TITLE
[cli & framework] Publishing size, upgrade_policy, reactivated e2e tests

### PR DESCRIPTION
**Pull request by wrwg**
### Description

This updates the CLI for `move publish` to specify which metadata to include (none, spare, or all). There is also a CLI-time check whether the users package become too large, and what to do about it.

Furtheremore, the `upgrade_policy` is now a field of the `Move.toml` package section, as it was always planned. This lead to some cascading changes throughout. Specifically, we had to duplicate some Move test packages to represent combinations of upgrade policies.

PR also adds some missing filed for code_publishing e2e tests, and reactivates those tests.

### Test Plan

Reactivated tests which were turned off.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3045)

### CODE REVIEW

File: crates/aptos/src/move_tool/mod.rs [hunk 1]
```diff
@@ -374,12 +436,20 @@ impl CliCommand<TransactionSummary> for PublishPackage {
...
                 bcs::to_bytes(&metadata).expect("PackageMetadata has BCS"),
                 compiled_units,
             );
+            let size = bcs::serialized_size(&payload)?;
+            if !override_size_check && size > MAX_PUBLISH_PACKAGE_SIZE {
+                return Err(CliError::UnexpectedError(format!(
+                    "The package is larger than 60k ({}k)! To lower the size \
+                you may want to include less artifacts via `--included_artifacts`. \
+                You can also override this check with `--override-size-check",
+                    size / 1000
```
Comment (gregnazario) [inline crates/aptos/src/move_tool/mod.rs:450]: I'd prefer if you just made both bytes so you could use the actual constant in the message.

File: aptos-move/framework/src/natives/code.rs [hunk 2]
```diff
@@ -71,7 +72,7 @@ pub struct UpgradePolicy {
 }
 
 impl UpgradePolicy {
-    pub fn no_compat() -> Self {
+    pub fn arbitrary() -> Self {
```
Comment (davidiw) [inline aptos-move/framework/src/natives/code.rs:75]: I guess resolve this arbitrary existence after devnet...

**Review by gregnazario — APPROVED** [review]
Lots of what feels like black magic and boilerplate code.  But looks good to me!

**Review by davidiw — APPROVED** [review]
the test for publishing is pretty gnarly, since it is the entirety of the move stdlib... do you think there's a smaller one that might be easier to parse?

**Comment by wrwg** [comment]
> the test for publishing is pretty gnarly, since it is the entirety of the move stdlib... do you think there's a smaller one that might be easier to parse?

This is the smallest possible test to verify framework upgradability, unfortunately, since Genesis installed the move-stdlib and everything which upgrades it must be a superset. There are already a couple of smaller tests for upgrade, but this one is still needed because of the special genesis situation, which initializes the metadata in a different way,

File: crates/aptos/src/move_tool/mod.rs (hunk 1 above)
Comment (wrwg) [reply crates/aptos/src/move_tool/mod.rs:450]: Using the constant now to construct the message.

**Pull request merged**